esp-mbedtls/src/lib.rs

```diff
@@ -279,35 +400,58 @@ pub struct Certificates<'a> {
     /// # Server:
     /// In server mode, this will be the certificate given to the client when
     /// performing a handshake.
-    pub certificate: Option<X509<'a>>,
+    certificate: Option<MbedTLSX509Crt>,
 
     /// Private key paired with the certificate. Must be set when [Certificates::certificate]
     /// is not [None]
-    pub private_key: Option<X509<'a>>,
-
-    /// Password used for the private key.
-    /// Use [None] when the private key doesn't have a password.
-    pub password: Option<&'a str>,
+    private_key: Option<PkContext>,
 }
 
-impl Default for Certificates<'_> {
+impl Default for Certificates {
     fn default() -> Self {
         Self::new()
     }
 }
 
-impl Certificates<'_> {
+impl Certificates {
     /// Create a new instance of [Certificates] with no certificates whatsoever
     pub const fn new() -> Self {
         Self {
             ca_chain: None,
             certificate: None,
             private_key: None,
-            password: None,
         }
     }
 
-    // Initialize the SSL using this set of certificates
+    /// Initialize the certificate the own certificate chain used for requests
+    ///
+    /// # Errors
+    ///
+    /// This function will fail with [TlsError::OutOfMemory] if there's not enough memory to
+    /// allocate certificates.
+    pub fn with_certificates(
```

Comment:
This would need two versions, just like `MbedTlsX509Crt` has two versions - one, in `Cetificates<'static>` which does copy, and another - in `Certificates<'d>` which tries to go non-copy where it can.